first-party/jib-classpath-main-class-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/classpathmainclass/ClasspathMainClassExtension.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.maven.extension.classpathmainclass;
+
+import com.google.cloud.tools.jib.api.buildplan.ContainerBuildPlan;
+import com.google.cloud.tools.jib.maven.extension.JibMavenPluginExtension;
+import com.google.cloud.tools.jib.maven.extension.MavenData;
+import com.google.cloud.tools.jib.plugins.extension.ExtensionLogger;
+import com.google.cloud.tools.jib.plugins.extension.JibPluginExtensionException;
+import java.util.Map;
+import java.util.Optional;
+
+public class ClasspathMainClassExtension implements JibMavenPluginExtension<Void> {
+
+  @Override
+  public Optional<Class<Void>> getExtraConfigType() {
+    return Optional.empty();
+  }
+
+  @Override
+  public ContainerBuildPlan extendContainerBuildPlan(
+      ContainerBuildPlan buildPlan,
+      Map<String, String> properties,
+      Optional<Void> extraConfig,
+      MavenData mavenData,
+      ExtensionLogger logger)
+      throws JibPluginExtensionException {
+    Map<String, String> newEnv = buildPlan.getEnvironment();
+    newEnv.putAll(ClasspathMainClassExtractor.extractEnvsFromEntrypoint(buildPlan.getEntrypoint()));
+    return ContainerBuildPlan.builder()
```

Comment:
```suggestion
    return buildPlan.toBuilder().setEnvironment(newEnv).build();
```
It's also possible to use `addEnvironment()`, but `setEnvironment()` looks simpler.